### ISSUE TITLE
Contradictory error and warning for `role` attribute in `summary` element

### ISSUE THREAD

**Issue by AlanDalton** [validator/validator#165]
Hi,

This validator works great. Thank you!

I’ve noticed a warning and error that contradict each other:

- When I validate HTML that includes a `summary` element with a `role` attribute, I get this warning:

> Warning: Element `summary` does not need a `role` attribute.

- When I validate HTML that includes a `summary` element with no `role` attribute, I get this error:

> Error: Element `summary` is missing one or more of the following attributes: `role`.

Regards,
Alan.

**Comment by sideshowbarker** [comment]
Please post a document or fragment that reproduces the problem. I can validate examples of `role`-less `summary` elements fine, and from examination of the validator sources I can say that in the general case that “Error: Element `summary` is missing one or more of the following attributes: `role`” is not going to be reported.

The only case in which I can imagine that message getting reported is if the `summary` has particular `aria-*` attributes that are only allowed when particular `role` values are also specified. In that case, the message is expected and conforms to the spec.

But I can’t confirm what the cause is for the message reported in this issue without having a document or fragment that reproduces the message that I can check.

**Comment by AlanDalton** [comment]
Yeah, you’re right. I had validated some code with a `summary` element that had an `aria-expanded` attribute but no `role` attribute. So, I tested this code:

```html
<!doctype html>
<html lang="en">
<meta charset="utf-8">
<title>Test summary role</title>
<details role="group" aria-labelledby="Test1">
	<summary role="button" aria-expanded="false" id="Test1">Test summary 1</summary>
	Test details 1.
</details>
<details role="group" aria-labelledby="Test2">
	<summary aria-expanded="false" id="Test2">Test summary 2</summary>
	Test details 2.
</details>
<details role="group" aria-labelledby="Test3">
	<summary id="Test3">Test summary 3</summary>
	Test details 3.
</details>
```
…and:
- `<summary role="button" aria-expanded="false" id="Test1">Test summary 1</summary>` causes this warning: “Element `summary` does not need a `role` attribute.”
- `<summary aria-expanded="false" id="Test2">Test summary 2</summary>` causes this error: “Element `summary` is missing one or more of the following attributes: `role`.”
- `<summary id="Test3">Test summary 3</summary>` causes no warnings or errors.

Sorry about that—and keep up the good work!

**Comment by sideshowbarker** [comment]
> Sorry about that—and keep up the good work!

No need to apologize—this stuff is confusing as hell to everybody
